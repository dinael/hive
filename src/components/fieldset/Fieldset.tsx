import { FC } from 'react'
import styles from './Fieldset.module.scss'
import { Label as LabelText } from '../label/Label'

// /**
// Uso de fieldset y legend
// Los elementos fieldset y legend de HTML se emplean para agrupar e identificar un conjunto de controles de formulario que están relacionados entre sí.

// Los lectores de pantalla normalmente leen el contenido de legend para cada control dentre de un fieldset, por lo que su contenido debería ser corto y descriptivo.

// Fieldset se debería utilizar únicamente para agrupar un conjunto de controles que necesitan una descripción para mejorar la accesibilidad, como por ejemplo un grupo de botones de radio o casillas de verificación.
// */

export type FieldsetProps = {
  legend?: string
  children?: any
  flat?: boolean
  required?: boolean
  isRequiredText?: string
  legendHidden?: boolean
  className?: string
}

export const Fieldset: FC<FieldsetProps> = ({
  legend,
  children,
  flat = false,
  required = false,
  isRequiredText,
  legendHidden = false,
  className,
  ...props
}: FieldsetProps) => {

  const fieldsetStyles = `
    ${styles['fieldset-component']}
    ${flat ? styles['flat'] : ''}
    ${className || ''}
  `

  return (
    <fieldset
      className={fieldsetStyles}
      {...props }>
      {legend ?
        <legend>
          <LabelText
            onlyText
            className={`${styles['inputfiled-label']}`}
            text={`${legend}`}
            required={required}
            isRequiredText={`${isRequiredText}`}
            labelHidden={legendHidden}/>
        </legend>
        : null}
        {children}
    </fieldset>
  )
}

export default Fieldset
