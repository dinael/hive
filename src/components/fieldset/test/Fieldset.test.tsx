import { describe, test, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import { Fieldset, FieldsetProps } from '../Fieldset'

describe('Fieldset', () => {
  const defaultProps: FieldsetProps = {
    legend: 'Legend Text',
    flat: false,
    required: true,
    isRequiredText: 'Required text',
    legendHidden: false,
    className: 'className-test'
  }

  test('renders snapshot', () => {
    const { container } = render(<Fieldset {...defaultProps} />)
    expect(container).toMatchSnapshot()
  })

  test('renders fieldset with correctly', () => {
    const { container } = render(<Fieldset {...defaultProps} />)
    const fieldsetElement = container.querySelector('fieldset')
    expect(fieldsetElement?.className).toContain('fieldset-component')
    expect(fieldsetElement?.tagName).toBe('FIELDSET')
    expect(container).toMatchSnapshot()
  })

  test('renders legend with correctly', () => {
    const { container } = render(<Fieldset {...defaultProps} />)
    const fieldsetElement = container.querySelector('fieldset > legend')
    expect(fieldsetElement?.tagName).toBe('LEGEND')
    expect(container).toMatchSnapshot()
  })


  test('renders with a custom class', () => {
    const { container } = render(<Fieldset {...defaultProps} />)
    const fieldsetElement = container.querySelector('fieldset')
    expect(fieldsetElement?.className).toContain(defaultProps.className)
    expect(container).toMatchSnapshot()
  })

  test('renders legend hidden', () => {
    const { container } = render(<Fieldset {...defaultProps} />)
    const fieldsetElement = container.querySelector('fieldset > legend')
    expect(fieldsetElement?.className).not.toContain('is-hidden')
    expect(container).toMatchSnapshot()
  })
})
