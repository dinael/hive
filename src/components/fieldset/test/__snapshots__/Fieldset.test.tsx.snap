// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Fieldset > Fieldset renders fieldset with correctly 1`] = `
<div>
  <fieldset
    class="
    fieldset-component
    
    className-test
  "
  >
    <legend>
      <span
        class="
    label-text-component
    
    is-required
    
    inputfiled-label
  "
      >
        Legend Text:
        <em
          class="required-text"
          title="Required text"
        >
          *
        </em>
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`Fieldset > Fieldset renders legend hidden 1`] = `
<div>
  <fieldset
    class="
    fieldset-component
    
    className-test
  "
  >
    <legend>
      <span
        class="
    label-text-component
    
    is-required
    
    inputfiled-label
  "
      >
        Legend Text:
        <em
          class="required-text"
          title="Required text"
        >
          *
        </em>
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`Fieldset > Fieldset renders legend with correctly 1`] = `
<div>
  <fieldset
    class="
    fieldset-component
    
    className-test
  "
  >
    <legend>
      <span
        class="
    label-text-component
    
    is-required
    
    inputfiled-label
  "
      >
        Legend Text:
        <em
          class="required-text"
          title="Required text"
        >
          *
        </em>
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`Fieldset > Fieldset renders snapshot 1`] = `
<div>
  <fieldset
    class="
    fieldset-component
    
    className-test
  "
  >
    <legend>
      <span
        class="
    label-text-component
    
    is-required
    
    inputfiled-label
  "
      >
        Legend Text:
        <em
          class="required-text"
          title="Required text"
        >
          *
        </em>
      </span>
    </legend>
  </fieldset>
</div>
`;

exports[`Fieldset > Fieldset renders with a custom class 1`] = `
<div>
  <fieldset
    class="
    fieldset-component
    
    className-test
  "
  >
    <legend>
      <span
        class="
    label-text-component
    
    is-required
    
    inputfiled-label
  "
      >
        Legend Text:
        <em
          class="required-text"
          title="Required text"
        >
          *
        </em>
      </span>
    </legend>
  </fieldset>
</div>
`;
